awscli/customizations/eks/update_kubeconfig.py

```diff
@@ -235,28 +235,29 @@ def _expand_path(self, path):
 
 
 class EKSClient(object):
-    def __init__(self, session, cluster_name, role_arn, parsed_globals=None):
+    def __init__(self, session, *, parsed_globals=None, parsed_args):
```

Comment:
Let's remove the `*` in the constructor and move the `parsed_args` to a required positional arg. I don't think we should be too worried about backwards compatibility here as this is the only module that relies on the `EKSClient` class. So by the end the initializer signature should be:
```python
def __init__(self, session, parsed_args, parsed_globals=None):
```